src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -438,42 +507,45 @@ winrt::guid Profile::GetGuidOrGenerateForJson(const Json::Value& json) noexcept
     return Profile::_GenerateGuidForProfile(name, source);
 }
 
-const HorizontalAlignment Profile::BackgroundImageHorizontalAlignment() const noexcept
-{
-    return std::get<HorizontalAlignment>(_BackgroundImageAlignment);
-}
-
-void Profile::BackgroundImageHorizontalAlignment(const HorizontalAlignment& value) noexcept
+#pragma region BackgroundImageAlignment
+bool Profile::HasBackgroundImageAlignment() const noexcept
 {
-    std::get<HorizontalAlignment>(_BackgroundImageAlignment) = value;
+    return _BackgroundImageAlignment.has_value();
 }
 
-const VerticalAlignment Profile::BackgroundImageVerticalAlignment() const noexcept
+void Profile::ClearBackgroundImageAlignment() noexcept
 {
-    return std::get<VerticalAlignment>(_BackgroundImageAlignment);
+    _BackgroundImageAlignment = std::nullopt;
 }
 
-void Profile::BackgroundImageVerticalAlignment(const VerticalAlignment& value) noexcept
+const HorizontalAlignment Profile::BackgroundImageHorizontalAlignment() const noexcept
 {
-    std::get<VerticalAlignment>(_BackgroundImageAlignment) = value;
+    const auto val{ _getBackgroundImageAlignmentImpl() };
+    return val ? std::get<HorizontalAlignment>(*val) : HorizontalAlignment::Center;
 }
 
-bool Profile::HasGuid() const noexcept
+void Profile::BackgroundImageHorizontalAlignment(const HorizontalAlignment& value) noexcept
 {
-    return _Guid.has_value();
+    if (!HasBackgroundImageAlignment() || std::get<HorizontalAlignment>(*_BackgroundImageAlignment) != value)
+    {
+        std::get<HorizontalAlignment>(*_BackgroundImageAlignment) = value;
+    }
 }
 
-winrt::guid Profile::Guid() const
+const VerticalAlignment Profile::BackgroundImageVerticalAlignment() const noexcept
 {
-    // This can throw if we never had our guid set to a legitimate value.
-    THROW_HR_IF_MSG(E_FAIL, !_Guid.has_value(), "Profile._guid always expected to have a value");
-    return *_Guid;
+    const auto val{ _getBackgroundImageAlignmentImpl() };
+    return val ? std::get<VerticalAlignment>(*val) : VerticalAlignment::Center;
 }
 
-void Profile::Guid(const winrt::guid& guid) noexcept
+void Profile::BackgroundImageVerticalAlignment(const VerticalAlignment& value) noexcept
 {
-    _Guid = guid;
+    if (!HasBackgroundImageAlignment() || std::get<VerticalAlignment>(*_BackgroundImageAlignment) != value)
+    {
+        std::get<VerticalAlignment>(*_BackgroundImageAlignment) = value;
+    }
 }
+#pragma endregion

```

Comment:
you can't do this.

the optional's contents are _UNDEFINED_ if it doesn't have a value. Remember last time? :smile: